src/symbolize/mod.rs

```diff
@@ -313,6 +313,22 @@ pub struct Sym<'src> {
 }
 
 
+/// The reason why symbolization failed.
+#[derive(Clone, Debug, PartialEq)]
+#[non_exhaustive]
+pub enum Reason {
+    /// The file offset does not map to a valid piece of code/data.
+    InvalidFileOffset,
+    /// The symbolization source has no or no relevant symbols.
+    ///
+    /// This reason could for instance be used if a shared object only
+    /// has dynamic symbols, but appears to be stripped aside from that.
+    MissingSyms,
+    /// XXX
+    UnknownAddr,
```

Comment:
Could we add something like "Failed to read file"? Can be in another diff.

Context: In AOSP sometimes we are unable to read files due to SELinux and it would be good if we can bubble up this issue.

Reply:
Have you actually seen this in the wild? Every actual issue should manifest as an error (`Err(blazesym::Error(...))`), including file opening due to permission denied ([example](https://github.com/libbpf/blazesym/blob/3c11f30c20757bff43e8bea9a20bb79972a14e9f/tests/blazesym.rs#L39-L42)). So I would expect that's what you would see.

Reply:
Scratch that. I was talking about the normalization API and this is about the symbolication. Sorry for the noise.

Reply:
With the rework of the normalization APIs (https://github.com/libbpf/blazesym/issues/321), there shouldn't be permission denied issues, because we no longer access all the ELF/APK files.